src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -361,21 +361,25 @@ def sync_deploy_root_with_stage(
 
         files_not_removed = []
         if local_paths_to_sync:
-            assert mapped_files is not None
+            assert bundle_map is not None
 
             # Deploying specific files/directories
             resolved_paths_to_sync = [
                 resolve_without_follow(p) for p in local_paths_to_sync
             ]
             if not recursive:
                 verify_no_directories(resolved_paths_to_sync)
-            deploy_paths_to_sync = [
-                source_path_to_deploy_path(p, mapped_files)
-                for p in resolved_paths_to_sync
-            ]
-            verify_exists(deploy_paths_to_sync)
+
+            deploy_paths_to_sync = []
+            for resolved_path in resolved_paths_to_sync:
```

Comment:
I went for minimal disturbance here because the focus was on `BundleMap` as a new abstraction. Once it's in, I'll rehash this logic to integrate better, but I'll likely need to tweak the APIs in the `BundleMap` class to do so. I didn't want the extra noise here.